Fix keyboard overlap with recent search items in the search tab

namanh-asher opened the pull request:
#### Summary
In the latest build, when I open the keyboard to input text on the tab search, it overlaps the content of the recent search item. This PR adds KeyboardAvoidingView to fix this issue.

#### Ticket Link
None

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
- Iphone 13 Simulator (iOS 16.1)
- Samsung S10 plus (Android 12)

#### Screenshots
Before fix:

https://github.com/mattermost/mattermost-mobile/assets/28721870/ee4da287-381d-4cb2-9062-f509d5d2a451


https://github.com/mattermost/mattermost-mobile/assets/28721870/e518c8b3-abeb-49f1-88b1-c05bf24a966f



After fix:

https://github.com/mattermost/mattermost-mobile/assets/28721870/af3f9d55-c724-4313-ab88-9f79e439c4cd


https://github.com/mattermost/mattermost-mobile/assets/28721870/9f3429f0-8398-4805-b22c-ab2955f1f22f


#### Release Note
```release-note
Fix keyboard overlap with recent search items in the search tab
```

mattermost-build commented:
Hello @namanh-asher,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

Per the [Mattermost Contribution Guide](https://developers.mattermost.com/contribute/getting-started/), we need to add you to the list of [approved contributors](https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true) for the Mattermost project.

**Please help complete the Mattermost [contribution license agreement](http://www.mattermost.org/mattermost-contributor-agreement/)?**
Once you have signed the CLA, please comment with `/check-cla` and confirm that the CLA check is green.

This is a standard procedure for many open source projects.

Please let us know if you have any questions.

We are very happy to have you join our growing community! If you're not yet a member, please consider joining our [Contributors community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors and discuss new opportunities with the core team.

namanh-asher commented:
/check-cla

enahum commented:
@namanh-asher thanks for the PR, can you please also share the same screen recordings but for Android?

namanh-asher commented:
> @namanh-asher thanks for the PR, can you please also share the same screen recordings but for Android?

@enahum Of course, here are the recordings with Android.
While recording, I discovered an issue when setting behavior='height'; Android loses two items. After that, I changed behavior to 'undefined,' and it works like the latest version on the Play Store. So, I updated the PR to set behavior to 'undefined'.

Android before fix:

https://github.com/mattermost/mattermost-mobile/assets/28721870/6c84993d-1610-4380-9b98-a2dc432779c9


Android with behavior="height":

https://github.com/mattermost/mattermost-mobile/assets/28721870/0165e965-694b-4bec-8d9f-ff19b0ad5b38

Android with behavior=undefined:

https://github.com/mattermost/mattermost-mobile/assets/28721870/3ca75f91-5bae-4e75-a15d-a309212824a1

larkox commented:
@yasserfaraazkhan please make sure you verify Android and iOS

mattermost-build commented:
Building app in separate branch.

yasserfaraazkhan commented:
Hi @namanh-asher ,
I tested this on 
- Android : User is able to see the recent search items without keyboard blocking it ✅ 
- On IOS 16: The searched items are still blocked by keyboard. I'm share a video. 
 

https://github.com/mattermost/mattermost-mobile/assets/15979783/d977c047-7688-43fe-8aaf-424965a4c6a9

namanh-bic commented:
> Hi @namanh-asher , I tested this on
> 
> * Android : User is able to see the recent search items without keyboard blocking it ✅
> * On IOS 16: The searched items are still blocked by keyboard. I'm share a video.
> 
>  RPReplay_Final1690368207.MP4

That's so weird. I just built it on iPhone 7 running iOS 15.7, and it worked fine, just like on the simulator I have tested. What iPhone do you have, and can you help me test it on another phone?

yasserfaraazkhan commented:
> That's so weird. I just built it on iPhone 7 running iOS 15.7, and it worked fine, just like on the simulator I have tested. What iPhone do you have, and can you help me test it on another phone?

 @namanh-asher , I'm using iPhone 12, iOS 16.6.

yasserfaraazkhan commented:
@namanh-asher 
I've checked it on iOS 16.4, I'm sharing a recording. can you try the this on your side?
https://github.com/mattermost/mattermost-mobile/assets/15979783/59b49835-4480-4dbb-9760-e2a81a52adb3

namanh-bic commented:
@yasserfaraazkhan, I just tested it on the iPhone SE 3 simulator, like in your screen recording, and it still works fine 😢. Could you please help me check the file app/screens/home/search/search.tsx to ensure that it has the KeyboardAvoidingView in your code?

https://github.com/mattermost/mattermost-mobile/assets/122859677/53c9912f-6b78-44d9-be0c-05105b4b6713

yasserfaraazkhan commented:
Hi @namanh-asher , I cleaned local and rebuild the branch. I can confirm its working on iOS.

https://github.com/mattermost/mattermost-mobile/assets/15979783/fe7f12dd-a864-48b1-81ec-122363a12943

yasserfaraazkhan reviewed (approved):
🎉

amyblais commented:
/cherry-pick release-2.7

mattermost-build commented:
Cherry pick is scheduled.
